pandas/core/internals.py

```diff
@@ -4708,15 +4709,20 @@ def _safe_reshape(arr, new_shape):
     return arr
 
 
-def _transform_index(index, func):
+def _transform_index(index, func, level=None):
     """
     Apply function to all values found in index.
 
     This includes transforming multiindex entries separately.
+    Only apply function to one level of the MultiIndex if level is specified.
 
     """
     if isinstance(index, MultiIndex):
-        items = [tuple(func(y) for y in x) for x in index]
+        if level is not None:
+            items = [tuple(func(y) if i == level else y
```

Comment:
So I mean this comment. Do you mean it would be cleaner to have a method on Index/MultiIndex itself that does this renaming? 
I can agree with that, but the problem is that there already is a 'rename' method which renames the `names` of a multi index (it could of course also be a private method)

But my preference is to leave the existing implementation intact in this PR (I just expanded the existing method a bit).